### PR TITLE
Change access folder structure

### LINKED ISSUE #3583: Grouping the extension access control contracts into one `extensions/` folder
**🧐 Motivation**
I think grouping the extension access control contracts would be more readable and organized, as it happens with the token standard contracts. 

**📝 Details**
### Current structure: 
```
contracts   
│
└─── access
    │   AccessControl.sol
    │   AccessControlCrossChain.sol     "Extension contract"
    │   AccessControlEnumerable.sol    "Extension contract"
    │   IAccessControlEnumerable.sol   "Extension contract interface"
    │   ...
    └───
```

### Expected structure: 
```
contracts   
│
└─── access
     │
     └─── extensions
     │    │   AccessControlCrossChain.sol     "Extension contract"
     │    │   AccessControlEnumerable.sol    "Extension contract"
     │    │   IAccessControlEnumerable.sol   "Extension contract interface"
     │    └───
     │   AccessControl.sol
     │   ...
     └───
```

### PR DESCRIPTION
**Pull request by RenanSouza2**
Fixes #3583

Changes access golder structure

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: .changeset/spicy-sheep-eat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': patch
```
Comment (ernestognw) [inline .changeset/spicy-sheep-eat.md:2]: I think it'd be major since it's a breaking change

```suggestion
'openzeppelin-solidity': major
```

**Review by ernestognw — COMMENTED** [review]
Thanks for the PR @RenanSouza2, we still need to agree on this, but in the meantime I'd recommend taking a look to the [docs preview](https://deploy-preview-4359--openzeppelin-contracts-docs.netlify.app/contracts/4.x/api/access), you can find them on the CI checks (Netlify), because there are a few things to address within this PR:

- The navigation for the `Access` section will likely change to reflect those changes
- There are some cross references to the access contracts across the documentation that we will need to make sure are not broken

**Comment by RenanSouza2** [comment]
> * There are some cross references to the access contracts across the documentation that we will need to make sure are not broken

About that, I am looking for broken links as so far the one I found was in the file `releases-satability.adoc` in [[major-releases]] section, the link is broken

**Review by Amxx — APPROVED** [review]

File: .changeset/spicy-sheep-eat.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': major
+---
+
+Change access folder structure
```
Comment (frangio) [inline .changeset/spicy-sheep-eat.md:5]:
```suggestion
`access`: Move `AccessControl` extensions to a dedicated directory.
```

**Comment by frangio** [comment]
I don't think this change breaks any links in the docs in particular.

**Pull request merged**